### DISCUSSION TITLE
Why is the global logger discouraged?

### DISCUSSION THREAD

**Discussion by jalaziz** [uber-go/zap#1283] [Q&A] [answered]
I keep seeing sources suggesting that the global logger should be avoided. For example, [this comment](https://github.com/uber-go/zap/issues/924#issuecomment-1207668887) and [this FAQ entry](https://github.com/uber-go/zap/blob/master/FAQ.md#why-include-package-global-loggers).

However, I can't seem to find an explanation as to _why_ it should be avoided.

One thing that comes to mind is that you lose some control of child loggers and their context. Why else is it discouraged?

**Comment by abhinav** [comment, marked as answer]
Hey @jalaziz.
Zap leans towards the philosophy of plumbing your dependencies explicitly where possible.
A global logger is provided, but we encourage that components that want to do any logging accept a logger as a constructor parameter and record it for later use. 

There are no known bugs/limitations with the global logger, if that's your concern. Your application will not misbehave if you use it.  Zap just believes that it's better application design to use an explicitly passed logger over the implicit global.

**Reply by jalaziz** [reply]
Thank you! I appreciate the clarification.
